master/internal/api_trials.go

```diff
@@ -912,6 +912,68 @@ func (a *apiServer) streamMetrics(ctx context.Context,
 	return nil
 }
 
+// Create a TrialSourceInfo, which serves as a link between trials and checkpoints
```

Comment:
since these aren't using any of `apiServer` properties, it'd be great to move them to `internal/trials/api_trials.go` which we used to have before [the trial collections removal](https://github.com/determined-ai/determined/pull/7297/files#diff-906e5e11d91f28bf7359060528b003d91b6b50f28491ebbbdafe83dc0a7bc388) and [will have again](https://github.com/determined-ai/determined/pull/7060/files#diff-906e5e11d91f28bf7359060528b003d91b6b50f28491ebbbdafe83dc0a7bc388) in detached mode?

basically: 
- add `type TrialsAPIServer struct{}`
- embed it into the apiserver in `api.go`
- move these methods onto `TrialsAPIServer`.